src/ftest.jl

```diff
@@ -229,3 +229,204 @@ function show(io::IO, ftr::FTestResult{N}) where N
     end
     print(io, '─'^totwidth)
 end
+
+
+##############################################
+# Tests of Between-Subjects Effects
+# Baset on F-statistics
+# L: The s×p full row rank matrix. The rows are estimable functions. s≥1 where p number of coefs
+
+struct GroupEffectsTable
+    name::Vector{String}
+    df::Vector{Float64}
+    f::Vector{Float64}
+    p::Vector{Float64}
+end
+
+"""
+θ + A * B * A'
+
+Change θ (only upper triangle). B is symmetric.
+"""
+function mulαβαtinc!(θ::AbstractMatrix, A::AbstractMatrix, B::AbstractMatrix)
+    axb  = axes(B, 1)
+    sa   = size(A, 1)
+    for m ∈ 1:sa
+        for n ∈ m:sa
+            for j ∈ axb
+                @inbounds for i ∈ axb
+                    θ[m, n] +=  A[m, i] * B[i, j] * A[n, j]
+                end
+            end
+        end
+    end
+    θ
+end
+"""
+a' * B * a
+
+"""
+function mulαtβα(a::AbstractVector, B::AbstractMatrix{T}) where T
+    if length(a) != size(B, 2)::Int  || size(B, 2)::Int  != size(B, 1)::Int  error("Dimention error") end
+    axbm  = axes(B, 1)
+    axbn  = axes(B, 2)
+    c = zero(T)
+    for i ∈ axbm
+        ct = zero(T)
+        for j ∈ axbn
+            @inbounds  ct += B[i, j] * a[j]
+        end
+        @inbounds c += ct * a[i]
+    end
+    c
+end
```

Comment:
FYI this is the same as `dot(a, B, a)` except significantly less efficient.

Reply:
change on `dot`